### PR TITLE
fix attributes discrepancies

### LINKED ISSUE #94: Discrepancies in attributes rendering
### empty style
```jsx
  <div
    style={{
      margin: null
    }}
  />
```

```
      + expected - actual

      -<div style="" data-reactroot="" data-reactid="1" data-react-checksum="179180076"></div>
      +<div data-reactroot="" data-reactid="1" data-react-checksum="1998851930"></div>
```

### multiple attribute in input
`<input multiple={false} />`

react: `<input/>`
rapscallion: `<input multiple="false"/>`

### custom attributes
`<input qwe={false} />`

react: `<input/>`
rapscallion: `<input qwe="false"/>`

### attributes order
```
 (client) ></svg></div><input type="file" accept="image/jpeg,image/png" style="display:none;"
 (server) ></svg></div><input accept="image/jpeg,image/png" type="file" style="display:none;"
```

It all maybe covered by #58

### PR DESCRIPTION
**Pull request by fleg**
picked attributes tests from #58 and fixed all of it, I use more of react-dom internals to minimize copy&paste and difference in result

also closes #94 

### wip
- [x] tests refactoring (merge attr.js and attributes.js)

-----

The maintainers of this repo require that all pull request submitters adhere to the following:

- [x] I have read and will comply with the [contribution guidelines](https://github.com/FormidableLabs/rapscallion/blob/master/CONTRIBUTE.md).
 _(required)_
- [x] I have read and will comply with the [code of conduct](https://github.com/FormidableLabs/rapscallion/blob/master/CONTRIBUTE.md).
 _(required)_
- [x] All related documentation has been updated to reflect the changes made. _(required)_
- [x] My commit messages are cleaned up and ready to merge. _(required)_

The maintainers of this repository require you to select the semantic version type that
the changes in this pull request represent.  Please select one of the following:
- [ ]  major
- [ ]  minor
- [x]  patch
- [ ]  documentation only

### CODE REVIEW

File: src/render/attrs.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+    if (markup) {
+      result += ` ${markup}`;
+    }
+  }
+
+  return result;
+}
+
+module.exports = renderAttrs;
```
Comment (divmain) [inline src/render/attrs.js]: Were changes introduced to this file, or was it just moved :( It's a pity GitHub has trouble displaying this in a helpful manner.

Comment (divmain) [inline src/render/attrs.js]: Looks like there are changes.  I'll have to pull this down and look at the diff directly.

**Comment by fleg** [comment]
rebased

**Comment by fleg** [comment]
refactor attributes test, but it still needs review

File: src/render/attrs.js (hunk 1 above)
Comment (fleg) [inline src/render/attrs.js]: A lot of changes actually :)

**Comment by fleg** [comment]
rebased
ping @divmain

**Comment by divmain** [comment]
Okay, I _finally_ pulled this down.  Thanks for all of your work on this!  This is ready to go, from my perspective.  There's a merge conflict, but as soon as that's fixed, this PR will be good to merge!

**Comment by fleg** [comment]
rebases, ci fail looks unrelated

**Pull request merged**